service-proxy.js

```diff
@@ -172,6 +172,24 @@ function rateLimit(req, buildRes) {
     var isExitNode = self.isExitFor(req.serviceName);
     if (isExitNode) {
         self.rateLimiter.createServiceCounter(req.serviceName);
+        self.rateLimiter.createKillSwitchCounter(req.serviceName);
+    }
+
+    // apply kill switch safe guard first
+    if (isExitNode && self.rateLimiter.shouldKillSwitchService(req.serviceName)) {
```

Comment:
we REALLY want this feature for total request.

Having a kill switch kick in when 2 * total limit would be great for defending co-tenant workers.